packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.js

```diff
@@ -12,13 +12,10 @@ const ldClientSideIdProduction = '5979d95f6040390cd07b5e01';
 // one above only for `production` environment.
 const ldClientSideIdStaging = '5979d95f6040390cd07b5e00';
 
-export const getFlopflipReconfiguration = defaultMemoize(projectKey => ({
```

Comment:
This is part of the problem. By memoizing here, the `ReconfigureFlopflip` never gets to invoke `componentDidUpdate` which makes it in some cases loose the fight for the latest props against `ConfigureFlopflip`.